assets/js/modules/reader-revenue-manager/datastore/publications.js

```diff
@@ -113,33 +116,23 @@ const baseActions = {
 			return;
 		}
 
-		const { onboardingState } = publication;
-		const currentOnboardingState = registry
-			.select( MODULES_READER_REVENUE_MANAGER )
-			.getPublicationOnboardingState();
-
-		const settings = registry
-			.select( MODULES_READER_REVENUE_MANAGER )
-			.getSettings();
-
-		if ( onboardingState !== currentOnboardingState ) {
-			settings.publicationOnboardingState = onboardingState;
-		}
-
-		// eslint-disable-next-line sitekit/no-direct-date
-		settings.publicationOnboardingStateLastSyncedAtMs = Date.now();
```

Comment:
As @aaemnnosttv kindly pointed out internally, we should not set the settings in this way as this apparently mutates the original settings object in registry. Rather, we should clone and create a new object, or just use `setSettings` directly as I've done in this PR.